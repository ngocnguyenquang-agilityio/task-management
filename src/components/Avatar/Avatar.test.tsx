import { render, screen } from '@testing-library/react';

// Components
import { Avatar } from '.';

const mockProps = {
  src: 'https://i.seadn.io/gae/gctNTZKmTRG5z7A56d1GOfh4pxaM_b-XtVrmFN4FE6269fZoIOhc5dtr4YVHaOGRiXkRBVTta91iuz344f6YpjCTda7sfOWC5qlp?auto=format&dpr=1&w=3840',
  name: 'test user',
};

describe('Avatar component', () => {
  test('Should match snapshot', () => {
    const comp = render(<Avatar {...mockProps} />);
    expect(comp).toMatchSnapshot();
  });

  test('Should has correct customClass', () => {
    render(<Avatar {...mockProps} customClass="custom-class" />);
    expect(screen.getByRole('img')).toHaveClass('custom-class');
  });

  test('Should render correct variant', () => {
    render(<Avatar {...mockProps} variant="circle" />);
    expect(screen.getByRole('img')).toHaveClass('rounded-full');
  });

  test('Should render fallback image', () => {
    const expectSrc = expect.stringContaining('ui-avatars.com');
    render(<Avatar name="test user" />);
    const image = screen.getByRole('img');
    expect(image).toHaveAttribute('srcset', expectSrc);
  });
});
