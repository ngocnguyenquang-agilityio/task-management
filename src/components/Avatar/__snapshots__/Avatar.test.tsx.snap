// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="test user avatar"
        class="rounded-lg"
        data-nimg="1"
        decoding="async"
        height="50"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fi.seadn.io%2Fgae%2FgctNTZKmTRG5z7A56d1GOfh4pxaM_b-XtVrmFN4FE6269fZoIOhc5dtr4YVHaOGRiXkRBVTta91iuz344f6YpjCTda7sfOWC5qlp%3Fauto%3Dformat%26dpr%3D1%26w%3D3840&w=128&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fi.seadn.io%2Fgae%2FgctNTZKmTRG5z7A56d1GOfh4pxaM_b-XtVrmFN4FE6269fZoIOhc5dtr4YVHaOGRiXkRBVTta91iuz344f6YpjCTda7sfOWC5qlp%3Fauto%3Dformat%26dpr%3D1%26w%3D3840&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fi.seadn.io%2Fgae%2FgctNTZKmTRG5z7A56d1GOfh4pxaM_b-XtVrmFN4FE6269fZoIOhc5dtr4YVHaOGRiXkRBVTta91iuz344f6YpjCTda7sfOWC5qlp%3Fauto%3Dformat%26dpr%3D1%26w%3D3840&w=128&q=75 2x"
        style="color: transparent;"
        width="50"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="test user avatar"
      class="rounded-lg"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fi.seadn.io%2Fgae%2FgctNTZKmTRG5z7A56d1GOfh4pxaM_b-XtVrmFN4FE6269fZoIOhc5dtr4YVHaOGRiXkRBVTta91iuz344f6YpjCTda7sfOWC5qlp%3Fauto%3Dformat%26dpr%3D1%26w%3D3840&w=128&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.seadn.io%2Fgae%2FgctNTZKmTRG5z7A56d1GOfh4pxaM_b-XtVrmFN4FE6269fZoIOhc5dtr4YVHaOGRiXkRBVTta91iuz344f6YpjCTda7sfOWC5qlp%3Fauto%3Dformat%26dpr%3D1%26w%3D3840&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fi.seadn.io%2Fgae%2FgctNTZKmTRG5z7A56d1GOfh4pxaM_b-XtVrmFN4FE6269fZoIOhc5dtr4YVHaOGRiXkRBVTta91iuz344f6YpjCTda7sfOWC5qlp%3Fauto%3Dformat%26dpr%3D1%26w%3D3840&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
